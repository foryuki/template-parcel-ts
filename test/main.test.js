const sao = require('sao')
const path = require('path')

const template = {
  fromPath: path.join(__dirname, '..')
}

describe('test', () => {
  it('ok', async () => {
    const stream = await sao.mockPrompt(template, {
      name: 'mockName',
      username: 'mockUsername',
      email: 'mockmail@mymail.com'
    })

    expect(stream.fileList).toMatchSnapshot()

    const pkg = JSON.parse(stream.fileContents('package.json'))
    expect(pkg.name).toBe('mockName')
    expect(pkg.author).toBe('mockUsername <mockmail@mymail.com>')
    expect(pkg.license).toBe('MIT')

    expect(stream.fileContents('.postcssrc.js')).toMatchSnapshot('.postcssrc.js')
    expect(stream.fileContents('README.md')).toMatchSnapshot('README.md')
    expect(stream.fileContents('.babelrc')).toMatchSnapshot('.babelrc')
    expect(stream.fileContents('tsconfig.json')).toMatchSnapshot('tsconfig.json')
  })
})

