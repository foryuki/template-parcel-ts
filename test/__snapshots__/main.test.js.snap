// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test ok 1`] = `
Array [
  ".babelrc",
  ".editorconfig",
  ".gitattributes",
  ".gitignore",
  ".npmrc",
  ".postcssrc",
  ".prettierrc",
  "LICENSE",
  "README.md",
  "commitlint.config.js",
  "index.html",
  "package.json",
  "src/assets/meme.png",
  "src/component/Title.tsx",
  "src/component/styles.scss",
  "src/index.tsx",
  "src/styles/styles.scss",
  "src/types/global.d.ts",
  "tsconfig.json",
  "tslint.json",
]
`;

exports[`test ok: .babelrc 1`] = `
"{
  \\"presets\\": [\\"env\\", \\"react\\"]
}
"
`;

exports[`test ok: .postcssrc 1`] = `
"{
  \\"modules\\": true,
  \\"plugins\\": {
    \\"autoprefixer\\": {
      \\"grid\\": true
    },
    \\"postcss-modules\\": {
      \\"generateScopedName\\": \\"[name]_[local]_[hash:base64:5]\\"
    }
  }
}
"
`;

exports[`test ok: README.md 1`] = `
"# mockName

## Installation
\`\`\`
// with npm
npm i

// with yarn
yarn
\`\`\`

## Run
\`\`\`bash
npm run start
\`\`\`

## License

MIT &copy; mockUsername
"
`;

exports[`test ok: tsconfig.json 1`] = `
"{
  \\"compilerOptions\\": {
    \\"baseUrl\\": \\".\\",
    \\"outDir\\": \\"dist\\",
    \\"module\\": \\"esnext\\",
    \\"target\\": \\"es5\\",
    \\"lib\\": [\\"es6\\", \\"dom\\"],
    \\"sourceMap\\": true,
    \\"allowJs\\": true,
    \\"jsx\\": \\"react\\",
    \\"moduleResolution\\": \\"node\\",
    \\"rootDir\\": \\"src\\",
    \\"forceConsistentCasingInFileNames\\": true,
    \\"noImplicitReturns\\": true,
    \\"noImplicitThis\\": true,
    \\"noImplicitAny\\": true,
    \\"importHelpers\\": true,
    \\"strictNullChecks\\": true,
    \\"suppressImplicitAnyIndexErrors\\": true,
    \\"noUnusedLocals\\": true,
    \\"allowSyntheticDefaultImports\\": true,
    \\"esModuleInterop\\": true
  },
  \\"exclude\\": [
    \\"node_modules\\"
  ]
}
"
`;
